server/client/interface.ts

```diff
@@ -0,0 +1,41 @@
+import { OpenSearchDashboardsClient } from "@opensearch-project/opensearch/api/opensearch_dashboards";
+import {
+  CoreSetup,
+  ILegacyCustomClusterClient,
+  LegacyCallAPIOptions,
+  Logger,
+  OpenSearchDashboardsRequest,
+  RequestHandlerContext,
+} from "opensearch-dashboards/server";
+
+export interface IRequestHandlerContentWithDataSource extends RequestHandlerContext {
```

Comment:
data source plugin registered client objects at https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/server/plugin.ts#L99 , not sure if that can be reused

Reply:
Good point and plugin should add dataSource plugin as requiredPlugins, add this change into this PR.